### PR TITLE
🦺 Add tree patch validator

### PR DESCRIPTION
**Pull request by SPGoding**
Split from #1152.

Warnings will be logged during development for missing command tree patches as suggested in https://github.com/SpyglassMC/Spyglass/pull/1152#issuecomment-2115270597

image

## Prerequisites

* #1190
* #1191

### CODE REVIEW

**Review by MulverineX — APPROVED** [review]

File: packages/java-edition/src/mcfunction/tree/patch-validator.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import type * as core from '@spyglassmc/core'
```
Comment (misode) [inline packages/java-edition/src/mcfunction/tree/patch-validator.ts:1]: The dash in this file name seems a bit odd, no? I would expect `patchValidator.ts`

Comment (MulverineX) [reply packages/java-edition/src/mcfunction/tree/patch-validator.ts:1]: Or patch_validator

**Review by misode — APPROVED** [review]

**Comment by MulverineX** [comment]
I will actually test this later, then I'll merge it

**Comment by MulverineX** [comment]
Yep, this is working! 👍

**Pull request merged**